curvefs/src/client/block_device_client.cpp

```diff
@@ -21,22 +21,199 @@
  * Author: xuchaojie
  */
 
+#include <string>
+
+#include "src/client/service_helper.h"
 #include "curvefs/src/client/block_device_client.h"
 
 namespace curvefs {
 namespace client {
 
+using ::curve::client::UserInfo;
+
 CURVEFS_ERROR BlockDeviceClientImpl::Init(
-    const BlockDeviceClientOptions &options) { return CURVEFS_ERROR::OK; }
+    const BlockDeviceClientOptions& options) {
+    if (fileClient_->Init(options.configPath) != LIBCURVE_ERROR::OK) {
+        return CURVEFS_ERROR::FAILED;
+    }
+
+    return CURVEFS_ERROR::OK;
+}
+
+void BlockDeviceClientImpl::UnInit() {
+    fileClient_->UnInit();
+}
+
+CURVEFS_ERROR BlockDeviceClientImpl::Open(const std::string& filename,
+                                          const std::string& owner) {
+    UserInfo userInfo(owner);
+    auto retCode = fileClient_->Open(filename, userInfo, nullptr);
+    if (retCode < 0) {
+        LOG(ERROR) << "Open file failed, filename = " << filename
+                   << ", retCode = " << retCode;
+        return CURVEFS_ERROR::FAILED;
+    }
+
+    fd_ = retCode;
+    isOpen_ = true;
+    filename_ = filename;
+    owner_ = owner;
+    return CURVEFS_ERROR::OK;
+}
+
+CURVEFS_ERROR BlockDeviceClientImpl::Close() {
+    if (!isOpen_ || fd_ < 0) {
+        LOG(WARNING) << "File already closed";
+        return CURVEFS_ERROR::BAD_FD;
```

Comment:
just return CURVEFS_ERROR::OK, no need to left error to up layer.

Reply:
> just return CURVEFS_ERROR::OK, no need to left error to up layer.

done